src/dist/replication/lib/replica_stub.cpp

```diff
@@ -2512,5 +2548,28 @@ void replica_stub::split_replica_error_handler(gpid pid, local_execution handler
     }
 }
 
+void replica_stub::update_disk_holding_replicas()
+{
+    for (const auto &dir_node : _fs_manager._dir_nodes) {
+        // clear the holding_primary_replicas/holding_secondary_replicas and re-calculate it from
+        // holding_replica
+        dir_node->holding_primary_replicas.clear();
+        dir_node->holding_secondary_replicas.clear();
+        for (auto iter = dir_node->holding_replicas.begin();
+             iter != dir_node->holding_replicas.end();
+             iter++) {
+            const std::set<dsn::gpid> &replicas_for_app = iter->second;
+            for (const auto &pid : replicas_for_app) {
+                replica_ptr r = get_replica(pid);
+                if (r->status() == partition_status::PS_PRIMARY) {
+                    dir_node->holding_primary_replicas[iter->first].emplace(pid);
+                } else if (r->status() == partition_status::PS_SECONDARY) {
+                    dir_node->holding_secondary_replicas[iter->first].emplace(pid);
+                }
```

Comment:
```suggestion
                } else {
                    dassert(r->status() == partition_status::PS_SECONDARY), "");
                    dir_node->holding_secondary_replicas[iter->first].emplace(pid);
                }
```

Reply:
`r->status() ` also contains `PS_POTENTIAL_SECONDARY` ,`PS_INVALID` and so on, this only focuses on `PS_PRIMARY` and `PS_SECONDARY`, `dassert ` may result coredump.